File: src/main/java/employeetech/com/employee/manager/controller/EmployeeResource.java
package employeetech.com.employee.manager.controller;

import employeetech.com.employee.manager.model.Employee;
import employeetech.com.employee.manager.service.EmployeeService;
import org.springframework.http.HttpStatus;
import org.springframework.http.HttpStatusCode;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("employee")
public class EmployeeResource {
    private final EmployeeService employeeService;

    public EmployeeResource(EmployeeService employeeService) {
        this.employeeService = employeeService;
    }

    @GetMapping("/all")
    public ResponseEntity<List<Employee>> getAllEmployees() {
        List<Employee> employees = employeeService.findAllEmployees();
        return ResponseEntity.status(HttpStatusCode.valueOf(200)).body(employees);
    }

    @GetMapping("find/{id}")
    public ResponseEntity<Employee> getEmployeeById(@PathVariable("id") Long id) {
        Employee employee = employeeService.findEmployeeById(id);
        return ResponseEntity.status(HttpStatusCode.valueOf(200)).body(employee);
    }

    @PostMapping("/add")
    public ResponseEntity<Employee> addEmployee(@RequestBody Employee employee) {
        Employee newEmployee = employeeService.addEmployee(employee);
        return ResponseEntity.status(HttpStatus.CREATED).body(newEmployee);
    }

    @PutMapping("/update/{id}")
    public ResponseEntity<Employee> updateEmployee(
            @PathVariable("id") Long id,
            @RequestBody Employee employee) {
        Employee updatedEmployee = employeeService.updateEmployee(id, employee);
        if (updatedEmployee != null) {
            return ResponseEntity.status(HttpStatus.OK).body(updatedEmployee);
        } else {
            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
        }
    }

    @DeleteMapping("/delete/{id}")
    public ResponseEntity<Void> deleteEmployee(@PathVariable("id") Long id) {
        employeeService.deleteEmployee(id);
        return ResponseEntity.status(HttpStatusCode.valueOf(200)).build();
    }
}

File: src/main/kotlin/employeetech/com/employee/manager/controller/EmployeeResource.kt
package employeetech.com.employee.manager.controller

import employeetech.com.employee.manager.model.Employee
import employeetech.com.employee.manager.service.EmployeeService
import org.springframework.http.HttpStatus
import org.springframework.http.HttpStatusCode
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("employee")
class EmployeeResource(private val employeeService: EmployeeService) {

    @GetMapping("/all")
    fun getAllEmployees(): ResponseEntity<List<Employee>> {
        val employees = employeeService.findAllEmployees()
        return ResponseEntity.status(HttpStatusCode.valueOf(200)).body(employees)
    }

    @GetMapping("find/{id}")
    fun getEmployeeById(@PathVariable("id") id: Long): ResponseEntity<Employee> {
        val employee = employeeService.findEmployeeById(id)
        return ResponseEntity.status(HttpStatusCode.valueOf(200)).body(employee)
    }

    @PostMapping("/add")
    fun addEmployee(@RequestBody employee: Employee): ResponseEntity<Employee> {
        val newEmployee = employeeService.addEmployee(employee)
        return ResponseEntity.status(HttpStatus.CREATED).body(newEmployee)
    }

    @PutMapping("/update/{id}")
    fun updateEmployee(
        @PathVariable("id") id: Long,
        @RequestBody employee: Employee
    ): ResponseEntity<Employee> {
        val updatedEmployee = employeeService.updateEmployee(id, employee)
        return if (updatedEmployee != null) {
            ResponseEntity.status(HttpStatus.OK).body(updatedEmployee)
        } else {
            ResponseEntity.status(HttpStatus.NOT_FOUND).build()
        }
    }

    @DeleteMapping("/delete/{id}")
    fun deleteEmployee(@PathVariable("id") id: Long): ResponseEntity<Unit> {
        employeeService.deleteEmployee(id)
        return ResponseEntity.status(HttpStatusCode.valueOf(200)).build()
    }
}
